### ISSUE TITLE
DC and Step current generator not working properly in NEURON for repeated runs

### ISSUE THREAD

**Issue by appukuttan-shailesh** [NeuralEnsemble/PyNN#487]
This test code can be [found here](https://github.com/appukuttan-shailesh/PyNN_testData/blob/master/issue487_step_dc.py).

NEURON output appears to vary depending on whether using:
(note: runtime = 100.0 ms; and that is where the issue arises)
``` python
run(runtime)
run(runtime)
```
Output:
![neuron_2runs](https://user-images.githubusercontent.com/24866517/27225772-fa44ca24-529b-11e7-9d07-09f6340fcf75.png)
or
``` python
run(runtime*2)
```
Output:
![neuron_1run](https://user-images.githubusercontent.com/24866517/27225780-ff53051c-529b-11e7-8135-61e4c180d9e6.png)

No such issues for NEST and Brian.

**Comment by apdavison** [comment]
I see the same behaviour

**Comment by appukuttan-shailesh** [comment]
Fixed in https://github.com/NeuralEnsemble/PyNN/pull/488

Updated output for first case in the earlier post:
```
run(runtime)
run(runtime)
```
![neuron_2runs_fixed](https://user-images.githubusercontent.com/24866517/27288570-7308dd66-5507-11e7-848c-032fd4991402.png)

**Comment by apdavison** [comment]
Fixed in #488
